app/forecasts/fetch.py

```diff
@@ -0,0 +1,210 @@
+        forecast.values.append(noon_forecast)
+    forecast.values.append(forecast_values)
+
+
+# pylint: disable=too-many-locals
+def _fetch_forecasts_by_stations(session, stations: List[WeatherStation]) -> List[WeatherForecast]:
+    """ Fetch forecasts for stations.
+    """
+    # Get the most recent model run:
+    most_recent_run = app.db.crud.get_most_recent_model_run(
+        session, app.db.crud.GDPS, app.db.crud.LATLON_15X_15)
+    # Get the predictions:
+    query = app.db.crud.get_model_run_predictions(
+        session, most_recent_run, map(lambda station: [station.long, station.lat], stations))
+
+    # Construct response object:
+    model_run = ModelRun(
+        datetime=most_recent_run.prediction_run_timestamp,
+        name=most_recent_run.prediction_model.name,
+        abbreviation=most_recent_run.prediction_model.abbreviation,
+        projection=most_recent_run.prediction_model.projection)
+
+    forecasts = []
+    tmp_station_list = stations.copy()
+    prev_grid = None
+    points = None
+    stations_in_polygon = None
+    forecasts_in_grid = {}
+
+    for grid, prediction in query:
+        if grid != prev_grid:
+            prev_grid = grid
+            forecasts_in_grid = {}
+            # Get the coordinates of the bounding geometry.
+            poly = to_shape(grid.geom)
+            # Get the bounding points (ignore the last point of the polygon)
+            points = list(poly.exterior.coords)[:-1]
+            # Match stations to the grid.
+            stations_in_polygon = _get_stations_for_polygon(
+                tmp_station_list, poly)
+            # Initialize forecasts for all the stations in this grid.
+            for station in stations_in_polygon:
+                forecast = WeatherForecast(
+                    station=station, model_run=model_run, values=[])
+                forecasts.append(forecast)
+                forecasts_in_grid[station.code] = forecast, NoonInterpolator()
+                # pop the station off the list
+                tmp_station_list.remove(station)
+
+        # It could conceivably happen that we have N where N>1 stations in a
+        # grid, in which case we need to iterate over the grid predictions N times.
+        for forecast, noon_interpolator in forecasts_in_grid.values():
+            _add_prediction_to_forecast(
+                forecast, prediction, points, noon_interpolator)
+        # NOTE: The code would be much simpler if we only did the interpolation afterwards.
+
+    return forecasts
+
+
+async def _fetch_forecasts_by_station_codes(station_codes: List[int]):
+    """ Fetch forecasts from database.
+    """
+    # Using the list of station codes, fetch the stations:
+    stations = await get_stations_by_codes(station_codes)
+    session = app.db.database.get_session()
+    # Fetch the all the forecasts
+    forecasts = _fetch_forecasts_by_stations(session, stations)
+
+    return forecasts
+
+
+async def fetch_forecasts(station_codes: List[int]):
```

Comment:
models

Reply:
"fetch_model_forecasts" - my thinking is - fetch models would return "gdps, rdps, whatever", as opposed to a "model forecast", e.g. GDPS forecast.